modules/imgcodecs/test/test_png.cpp

```diff
@@ -7,6 +7,19 @@ namespace opencv_test { namespace {
 
 #if defined(HAVE_PNG) || defined(HAVE_SPNG)
 
+TEST(imRead, passing_Mat)
+{
+  Mat img = Mat::zeros(1000, 1000, CV_8UC3);
+  Mat subImg = img.rowRange(64, 576).colRange(128, 640);
+  imread(samples::findFile("/cv/shared/lena.png"), subImg);
+  Mat ori = imread(samples::findFile("/cv/shared/lena.png"));
+  EXPECT_EQ(0, cv::norm(ori, subImg, NORM_INF));
```

Comment:
You have to compare with img ROI again. `subImg` can be reallocated and test won't catch this.

```cpp
Mat img = Mat::zeros(1000, 1000, CV_8UC3);
Mat subImg = img.rowRange(64, 576).colRange(128, 640);
imread(samples::findFile("/cv/shared/lena.png"), subImg);

subImg = img.rowRange(64, 576).colRange(128, 640);
Mat ori = imread(samples::findFile("cv/shared/lena.png"));
EXPECT_EQ(0, cv::norm(ori, subImg, NORM_INF));
```

Reply:
@thewoz , I refered this comment for your question in https://github.com/opencv/opencv/pull/24415#issuecomment-1827506930 on how to use `findFile` and what is a correct path to test image.

Reply:
ah ok sorry

Reply:
Hi @dkurt,

I've tried three different ways, and I don't understand what the problem is.

NOTE: the $OPENCV_TEST_DATA_PATH is set in the correct way

**- [1] samples::findFile("samples/data/lena.jpg")**

I got this warning but eventually the image is loaded correctly

`[ WARN:0@0.435] global samples.cpp:61 findFile cv::samples::findFile('samples/data/lena.jpg') => '/Users/thewoz/Dropbox/Research/Github/opencv1/build/..//samples/data/lena.jpg'`

**- [2] findDataFile("cv/shared/lena.png")**

I got this error:

`unknown file: Failure
C++ exception with description "OpenCV(4.8.0-dev) /Users/thewoz/Dropbox/Research/Github/opencv1/modules/ts/src/ts.cpp:1071: error: (-2:Unspecified error) OpenCV tests: Can't find required data file: cv/shared/lena.png in function 'findData'
" thrown in the test body.`

this I do not understand why it does not work as it is called in other files:

modules/dnn/test/test_caffe_importer.cpp:    Mat img = imread(findDataFile("cv/shared/lena.png"));
modules/dnn/test/test_tflite_importer.cpp:    Mat input = imread(findDataFile("cv/shared/lena.png"));
modules/dnn/test/test_tflite_importer.cpp:    Mat input = imread(findDataFile("cv/shared/lena.png"));
modules/dnn/test/test_torch_importer.cpp:    Mat sample = imread(findDataFile("cv/shared/lena.png"));

**- [3] samples::findFile("cv/shared/lena.png")**

I got this:

`[ WARN:0@0.437] global samples.cpp:61 findFile cv::samples::findFile('cv/shared/lena.png') => ''
unknown file: Failure
C++ exception with description "OpenCV(4.8.0-dev) /Users/thewoz/Dropbox/Research/Github/opencv1/modules/core/src/utils/samples.cpp:64: error: (-2:Unspecified error) OpenCV samples: Can't find required data file: cv/shared/lena.png in function 'findFile'
" thrown in the test body.`

I could not find any other examples of calls to "cv::samples::findFile('cv/xxxxx') in the code.
I understand that cv::samples::findFile() cerca nel OPENCV_TEST_DATA_PATH

How can I solve it?
sorry so much for the trouble

Reply:
C++ Test passed on CI: https://pullrequest.opencv.org/buildbot/builders/precommit_linux64/builds/105434

now just make Python happy (replace cv2 usage to cv)

Reply:
Sorry but Python is not my favorite language :)
This is how I set the environment variable:
export OPENCV_TEST_DATA_PATH=/Users/thewoz/Github/opencv_extra/testdata/
It's correct?